scrape/scrape.go

```diff
@@ -1906,3 +1921,18 @@ func TargetFromContext(ctx context.Context) (*Target, bool) {
 	t, ok := ctx.Value(ctxKeyTarget).(*Target)
 	return t, ok
 }
+
+func pickSchema(bucketFactor float64) int32 {
+	if bucketFactor <= 1 {
+		bucketFactor = 1
+	}
+	floor := math.Floor(math.Log2(math.Log2(bucketFactor)))
```

Comment:
This currently gives 4 for factor 1.09 now. The reason is that when you get to 
`-schema <= log(log(factor))`

the equation reverses the operator when you multiply both sides by -1
`schema >= -log(log(factor))`

So actually you need to calculate floor on the negative value I think to get the right rounding.
```suggestion
	floor := math.Floor(-math.Log2(math.Log2(bucketFactor)))
```

Also the benefit is that you don't need to negate the result in the switch.

Given this, I think there should be a unit test for the table that we documented.